Source/JavaScriptCore/runtime/JSGlobalObject.h

```diff
@@ -1194,9 +1254,13 @@ class JSGlobalObject : public JSSegmentedVariableObject {
     void installMapPrototypeWatchpoint(MapPrototype*);
     void installSetPrototypeWatchpoint(SetPrototype*);
     void tryInstallArrayBufferSpeciesWatchpoint(ArrayBufferSharingMode);
+    void tryInstallTypedArraySpeciesWatchpoint(TypedArrayType);
+    void installTypedArrayConstructorSpeciesWatchpoint(JSTypedArrayViewConstructor*);
 
 protected:
-    void tryInstallSpeciesWatchpoint(JSObject* prototype, JSObject* constructor, std::unique_ptr<ObjectPropertyChangeAdaptiveWatchpoint<InlineWatchpointSet>>& constructorWatchpoint, std::unique_ptr<ObjectPropertyChangeAdaptiveWatchpoint<InlineWatchpointSet>>& speciesWatchpoint, InlineWatchpointSet& speciesWatchpointSet);
+    enum class SpeciesEmpty : bool { Yes, No };
```

Comment:
nit: I think I'd name this "HasSpecies" or "HasSpeciesProperty", and invert the uses of it

Reply:
Changed.